src/main/scala/mesosphere/marathon/storage/repository/GcActor.scala

```diff
@@ -79,7 +81,22 @@ private[storage] class GcActor[K, C, S](
   private var lastCompactStart = Instant.now()
   private val compactTime = Kamon.metrics.histogram("GarbageCollector.compactTime", Time.Milliseconds)
 
-  startWith(Idle, IdleData)
+  if (zkCleaningInterval == 0.millis) {
+    startWith(Idle, IdleData)
+  } else {
+    startWith(Resting, IdleData)
+  }
+
+  when(Resting) {
+    case Event(WakeUp, _) =>
+      goto(Idle) using IdleData
```

Comment:
Hm, that is a little confusing. Why are we going to idle?

Reply:
Yeah I think it's super hard to see distinction between Idle and Resting. I think we need new names... Idle -> AwaitingGC, Resting -> Idle? Maybe?

Also don't forget to adjust the javadoc for this file that describes the state